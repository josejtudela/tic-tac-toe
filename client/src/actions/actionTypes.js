const actionTypes = {
  NEW: 'NEW_GAME',
  MOVE: 'MOVE_PLAYER',
};

export default actionTypes;
