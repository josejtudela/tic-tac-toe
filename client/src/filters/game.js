const findById = (state, id) => {
  if (state === undefined) {
    return state;
  }
  const squareData = state.squares.filter((square) => {
    const isSameId = square.id === id;
    return isSameId;
  })[0];

  return squareData;
};

export default findById;
