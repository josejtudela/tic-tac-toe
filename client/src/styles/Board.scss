$break-small: 600px;
$break-large: 1200px;

.gridBoard {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
}

.gridContainer{
    width: 35vw;
    height: 35vw;
    background-color: tomato;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    /* grid-template-rows: repeat(2, minmax(16em,auto)); */
    grid-gap: 1em;
    padding: 20px;
    border-radius: 5px;
    box-shadow: 0 4px 6px 0 hsla(0, 0%, 0%, 0.2);
    @media screen and (max-width: $break-large) {
        width: 45vw;
        height: 45vw;
        transition: 1s;
    }
    @media screen and (max-width: $break-small) {
        width: 80vw;
        height: 80vw;
        transition: 1s;
    }

}

.gridPlayerWin{
    width: 40vw;
    height: 40vw;
    background-color: tomato;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    padding: 20px;
    border-radius: 5px;
    box-shadow: 0 4px 6px 0 hsla(0, 0%, 0%, 0.2);
    transition: 2s;
    text-align: center;
    @media screen and (max-width: $break-large) {
        width: 45vw;
        height: 45vw;
        transition: 1s;
    }
    @media screen and (max-width: $break-small) {
        width: 80vw;
        height: 80vw;
        transition: 1s;
    }
}

.buttonBoard {
    background-color: white;
    font-size: 1.2em;
    cursor: pointer;
    color: tomato;
    padding: 0.5em;
    font-weight: 700;
    border: 0px;
    border-radius: 4px;
    margin-top: 1em;
    &:hover {
        background-color: rgb(218, 210, 210);
    }
}

.buttonBoardWin {
    background-color: tomato;
    font-size: 1.2em;
    cursor: pointer;
    color: rgb(243, 243, 243);
    padding: 0.5em;
    font-weight: 700;
    border: 0px;
    border-radius: 4px;
    margin-top: 1em;
    &:hover {
        background-color: rgba(255, 99, 71, 0.692);
    }
}