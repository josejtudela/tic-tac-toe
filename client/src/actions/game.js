import actionTypes from './actionTypes';

export function movePlayer(squareId) {
  // if (squareId === undefined || squareId === '') return { type: '' };

  return {
    type: actionTypes.MOVE,
    squareId,
  };
}

export function newGame() {
  return {
    type: actionTypes.NEW,
  };
}
