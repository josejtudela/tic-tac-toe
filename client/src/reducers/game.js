import uuid from 'uuid';
import checkWin from '../utils/checkWinners';
import actionTypes from '../actions/actionTypes';

const squares = [];
for (let i = 0; i <= 8; i += 1) {
  squares.push({ id: uuid(), value: 0 });
}

export const initialState = {
  squares,
  turnPlayer: 1,
  playerWin: false,
};

function reducer(state = initialState, action) {
  let newSquares = [];
  switch (action.type) {
    case actionTypes.MOVE:
      newSquares = state.squares.map((square) => {
        if (square.id === action.squareId) {
          return {
            id: square.id,
            value: state.turnPlayer,
          };
        }
        return {
          ...square,
        };
      });

      return {
        ...state,
        turnPlayer: state.turnPlayer === 1 ? 2 : 1,
        playerWin: checkWin(newSquares),
        squares: newSquares,
      };
    case actionTypes.NEW:
      newSquares = state.squares.map(square => ({ ...square, value: 0 }));
      return {
        ...state,
        turnPlayer: 1,
        playerWin: false,
        squares: newSquares,
      };
    default:
      break;
  }
  return state;
}

export default reducer;
