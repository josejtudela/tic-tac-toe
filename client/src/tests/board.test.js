import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Board } from '../components/Board.component';
import { initialState } from '../reducers/game';
import { newGame } from '../actions/game';

configure({ adapter: new Adapter() });

describe('<Board />', () => {
  test('Should be render render with default squares', () => {
    const mockOnHandleReset = jest.fn();
    const wrapper = shallow(
      <Board  
        squares={initialState.squares}
        turnPlayer={initialState.turnPlayer}
        playerWin={initialState.playerWin}
        onHandleReset={mockOnHandleReset}
      />)

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('h2').text()).toBe('Player One');
      expect(wrapper.find('button').text()).toBe('Reset');
  });

  test('Should be render render with one square setup', () => {
    const mockOnHandleReset = jest.fn();
    initialState.squares[0].value = 1;
    initialState.turnPlayer = 2;
    const wrapper = shallow(
      <Board  
        squares={initialState.squares}
        turnPlayer={initialState.turnPlayer}
        playerWin={initialState.playerWin}
        onHandleReset={mockOnHandleReset}
      />)

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('h2').text()).toBe('Player Two');
      expect(wrapper.find('button').text()).toBe('Reset');
  });
});

describe('Handle Button', () => {
  test('simulate click button reset', () => {
    const mockOnHandleReset = jest.fn(newGame);
    const wrapper = shallow(
      <Board  
        squares={initialState.squares}
        turnPlayer={initialState.turnPlayer}
        playerWin={initialState.playerWin}
        onHandleReset={mockOnHandleReset}
      />)

    wrapper.find('button').simulate('click');
    expect(mockOnHandleReset).toHaveBeenCalled();
  });
});

