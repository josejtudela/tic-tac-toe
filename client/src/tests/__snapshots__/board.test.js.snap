// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> Should be render render with default squares 1`] = `
<div
  className="gridBoard"
>
  <div>
    <h2>
      Player One
    </h2>
    <div
      className="gridContainer"
    >
      <Connect(Component)
        key="id0"
        squareId="id0"
      />
      <Connect(Component)
        key="id1"
        squareId="id1"
      />
      <Connect(Component)
        key="id2"
        squareId="id2"
      />
      <Connect(Component)
        key="id3"
        squareId="id3"
      />
      <Connect(Component)
        key="id4"
        squareId="id4"
      />
      <Connect(Component)
        key="id5"
        squareId="id5"
      />
      <Connect(Component)
        key="id6"
        squareId="id6"
      />
      <Connect(Component)
        key="id7"
        squareId="id7"
      />
      <Connect(Component)
        key="id8"
        squareId="id8"
      />
    </div>
  </div>
  <button
    className="buttonBoard"
    onClick={[Function]}
    type="button"
  >
    Reset
  </button>
</div>
`;

exports[`<Board /> Should be render render with one square setup 1`] = `
<div
  className="gridBoard"
>
  <div>
    <h2>
      Player Two
    </h2>
    <div
      className="gridContainer"
    >
      <Connect(Component)
        key="id0"
        squareId="id0"
      />
      <Connect(Component)
        key="id1"
        squareId="id1"
      />
      <Connect(Component)
        key="id2"
        squareId="id2"
      />
      <Connect(Component)
        key="id3"
        squareId="id3"
      />
      <Connect(Component)
        key="id4"
        squareId="id4"
      />
      <Connect(Component)
        key="id5"
        squareId="id5"
      />
      <Connect(Component)
        key="id6"
        squareId="id6"
      />
      <Connect(Component)
        key="id7"
        squareId="id7"
      />
      <Connect(Component)
        key="id8"
        squareId="id8"
      />
    </div>
  </div>
  <button
    className="buttonBoard"
    onClick={[Function]}
    type="button"
  >
    Reset
  </button>
</div>
`;
